src/probnum/statespace/sde.py

```diff
@@ -298,6 +353,101 @@ def f(t, y):
 
         return f, y0
 
+    def _setup_vectorized_mde_forward_sqrt(self, initrv, _diffusion=1.0):
+        r"""Set up forward moment differential equations (MDEs) using a square-root
+        implementation. (https://ieeexplore.ieee.org/document/4045974)
+
+        The covariance :math:`P(t)` obeys the Riccati equation
+
+        .. math::
+            \dot P(t) = G(t)P(t) + P(t)G^\top(t) + L(t)L^\top(t).
+
+        Let :math:`S(t)` be a square-root of :math:`P(t)`, :math:`P(t)` positive definite, then
+
+        .. math::
+            P(t) = S(t)S^\top(t)
+
+        and we get the Riccati-Equation
+
+        .. math::
+            \dot P(t) = G(t)S(t)S^\top(t) + 1/2 \cdot L(t)L^\top(t)S^{-\top}S^\top
+                        + S(t)S^\top(t)G^\top(t) + 1/2 \cdot S(t)S^{-1}(t)L(t)L^\top(t).
+
+        One solution can be found by the square-root :math:`\dot S(t)`
+
+        .. math::
+            \dot S(t) = G(t)S(t) + (A + 1/2 \cdot L(t)L^\top(t))S^{-\top}
+
+        where :math:`A` is an arbitrary symmetric matrix.
+        :math:`A` can be chosen to make S lower-triangular which can be achieved by
+
+        .. math::
+            M(t) = S^{-1}(t)\dot S(t) + \dot S(t)^\top S^{-\top}
+
+        and
+
+        .. math::
+            M(t) = \bar G(t) + \bar G^\top(t) + \bar L(t) \bar L^\top(t)
+
+        and
+
+        .. math::
+            \bar G(t) = S^{-1}(t)G(t)S(t),
+            \bar L(t) = S^{-1}L(t)
+
+        and
+
+        .. math::
+            \dot S(t) = S(t)[M(t)]_{\mathrm{lt}}
+
+        where :math:`\mathrm{lt}` denotes the lower-triangular operator defined by
+
+        .. math::
+            [M(t)]{_{\mathrm{lt}}}_{ij} =
+                \begin{cases}
+                    0 & i < j\\
+                    1/2 m(t)_{ij} & i=j\\
+                    m(t)_{ij} & i > j
+                \end{cases}.
+
+        Compute an ODE vector field that represents the MDEs and is
+        compatible with scipy.solve_ivp.
+        """
+        dim = len(initrv.mean)
+
+        def f(t, y):
+            # Undo vectorization
+            mean, cov_cholesky_flat = y[:dim], y[dim:]
+            cov_cholesky = cov_cholesky_flat.reshape((dim, dim))
+
+            # Apply iteration
+            G = self.driftmatfun(t)
+            u = self.forcevecfun(t)
+            L = self.dispmatfun(t)
+
+            new_mean = G @ mean + u
+            G_bar = scipy.linalg.solve_triangular(
+                cov_cholesky, G @ cov_cholesky, lower=True
+            )
+            L_bar = np.sqrt(_diffusion) * scipy.linalg.solve_triangular(
+                cov_cholesky, L, lower=True
+            )
+            M = G_bar + G_bar.T + L_bar @ L_bar.T
+
+            new_cov_cholesky = cholesky_update(
```

Comment:
I think you meant `utils.linalg.tril_to_positive_tril` here?!

Reply:
I think this should result in the same matrices, right? But I see that using `utils.linalg.tril_to_positive_tril` might be more descriptive of this use case (and we'd save the call to `np.linalg.qr`).

Reply:
yes, the call to qr is the things that should be avoided here. QR decompositions are expensive :)